### ISSUE TITLE
 java.lang.NoSuchMethodError: com.sksamuel.elastic4s.indexes.CreateIndexContentBuilder

### ISSUE THREAD

**Issue by arushijain** [sksamuel/elastic4s#982]
Dependencies 

```   
    val elastic4sVersion = "5.4.0"

    // Elastic Search
    "com.sksamuel.elastic4s" %% "elastic4s-core" % Versions.elastic4sVersion,
    // for the tcp client
    "com.sksamuel.elastic4s" %% "elastic4s-tcp" % Versions.elastic4sVersion,

    // for the http client
    "com.sksamuel.elastic4s" %% "elastic4s-http" % Versions.elastic4sVersion,

    // if you want to use reactive streams
    "com.sksamuel.elastic4s" %% "elastic4s-streams" % Versions.elastic4sVersion,

    // testing
    "com.sksamuel.elastic4s" %% "elastic4s-testkit" % Versions.elastic4sVersion % "test",
    "com.sksamuel.elastic4s" %% "elastic4s-embedded" % Versions.elastic4sVersion % "test",
    "com.sksamuel.elastic4s" % "elastic4s-json4s_2.11" % "6.0.0-M1",
    "com.sksamuel.elastic4s" % "elastic4s-circe_2.11" % "6.0.0-M1"

```

Code Executed 

```

import com.sksamuel.elastic4s.http.ElasticDsl._
import com.sksamuel.elastic4s.{ElasticsearchClientUri}
import com.sksamuel.elastic4s.http.HttpClient

 val host = config.getString("localhost")
      val port = config.getInt(9200)
      HttpClient(ElasticsearchClientUri(host, port))

      client.execute {
        createIndex("test")
          .mappings(
          mapping("maturity").as(
            textField("name")
            )
        )
      }.await

```

Error

```
[info]   ...
java.lang.NoSuchMethodError: com.sksamuel.elastic4s.indexes.CreateIndexContentBuilder$.apply(Lcom/sksamuel/elastic4s/indexes/CreateIndexDefinition;)Lorg/elasticsearch/common/xcontent/XContentBuilder;
	at com.sksamuel.elastic4s.http.index.IndexAdminImplicits$CreateIndexExecutable$.execute(IndexAdminImplicits.scala:156)
	at com.sksamuel.elastic4s.http.index.IndexAdminImplicits$CreateIndexExecutable$.execute(IndexAdminImplicits.scala:146)
	at com.sksamuel.elastic4s.http.HttpClient$class.execute(HttpClient.scala:23)
	at com.sksamuel.elastic4s.http.HttpClient$$anon$1.execute(HttpClient.scala:37)
	at com.neighborly.Storage$Test$$anonfun$3$$anonfun$apply$mcV$sp$21.apply(Storage$Test.scala:213)
	at com.neighborly.Storage$Test$$anonfun$3$$anonfun$apply$mcV$sp$21.apply(Storage$Test.scala:212)
	at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
	at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
	at org.scalatest.Transformer.apply(Transformer.scala:22)
	at org.scalatest.Transformer.apply(Transformer.scala:20)
	at org.scalatest.FunSpecLike$$anon$1.apply(FunSpecLike.scala:454)
	at org.scalatest.TestSuite$class.withFixture(TestSuite.scala:196)
	at org.scalatest.FunSpec.withFixture(FunSpec.scala:1630)
	at org.scalatest.FunSpecLike$class.invokeWithFixture$1(FunSpecLike.scala:451)
	at org.scalatest.FunSpecLike$$anonfun$runTest$1.apply(FunSpecLike.scala:464)
	at org.scalatest.FunSpecLike$$anonfun$runTest$1.apply(FunSpecLike.scala:464)
	at org.scalatest.SuperEngine.runTestImpl(Engine.scala:289)
	at org.scalatest.FunSpecLike$class.runTest(FunSpecLike.scala:464)
	at org.scalatest.FunSpec.runTest(FunSpec.scala:1630)
	at org.scalatest.FunSpecLike$$anonfun$runTests$1.apply(FunSpecLike.scala:497)
	at org.scalatest.FunSpecLike$$anonfun$runTests$1.apply(FunSpecLike.scala:497)
	at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:396)
	at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:384)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:384)
	at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:373)
	at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:410)
	at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:384)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:384)
	at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:379)
	at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:461)
	at org.scalatest.FunSpecLike$class.runTests(FunSpecLike.scala:497)
	at org.scalatest.FunSpec.runTests(FunSpec.scala:1630)
	at org.scalatest.Suite$class.run(Suite.scala:1147)
	at org.scalatest.FunSpec.org$scalatest$FunSpecLike$$super$run(FunSpec.scala:1630)
	at org.scalatest.FunSpecLike$$anonfun$run$1.apply(FunSpecLike.scala:501)
	at org.scalatest.FunSpecLike$$anonfun$run$1.apply(FunSpecLike.scala:501)
	at org.scalatest.SuperEngine.runImpl(Engine.scala:521)
	at org.scalatest.FunSpecLike$class.run(FunSpecLike.scala:501)
	at org.scalatest.FunSpec.run(FunSpec.scala:1630)
	at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:314)
	at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:480)
	at sbt.TestRunner.runTest$1(TestFramework.scala:76)
	at sbt.TestRunner.run(TestFramework.scala:85)
	at sbt.TestFramework$$anon$2$$anonfun$$init$$1$$anonfun$apply$8.apply(TestFramework.scala:202)
	at sbt.TestFramework$$anon$2$$anonfun$$init$$1$$anonfun$apply$8.apply(TestFramework.scala:202)
	at sbt.TestFramework$.sbt$TestFramework$$withContextLoader(TestFramework.scala:185)
	at sbt.TestFramework$$anon$2$$anonfun$$init$$1.apply(TestFramework.scala:202)
	at sbt.TestFramework$$anon$2$$anonfun$$init$$1.apply(TestFramework.scala:202)
	at sbt.TestFunction.apply(TestFramework.scala:207)
	at sbt.Tests$.sbt$Tests$$processRunnable$1(Tests.scala:239)
	at sbt.Tests$$anonfun$makeSerial$1.apply(Tests.scala:245)
	at sbt.Tests$$anonfun$makeSerial$1.apply(Tests.scala:245)
	at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:44)
	at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:44)
	at sbt.std.Transform$$anon$4.work(System.scala:63)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
	at sbt.Execute.work(Execute.scala:237)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
	at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)

```

**Comment by arushijain** [comment]
When I use `indexInto` I get the same error

```

val res = client.execute {
       indexInto("test" / "info") doc Info("tests1")
 }.await

java.lang.NoSuchMethodError: com.sksamuel.elastic4s.indexes.IndexContentBuilder$.apply(Lcom/sksamuel/elastic4s/indexes/IndexDefinition;)Lorg/elasticsearch/common/xcontent/XContentBuilder;
  at com.sksamuel.elastic4s.http.index.IndexImplicits$IndexHttpExecutable$.execute(IndexImplicits.scala:35)
  at com.sksamuel.elastic4s.http.index.IndexImplicits$IndexHttpExecutable$.execute(IndexImplicits.scala:12)
  at com.sksamuel.elastic4s.http.HttpClient$class.execute(HttpClient.scala:23)
  at com.sksamuel.elastic4s.http.HttpClient$$anon$1.execute(HttpClient.scala:37)
  ... 45 elided

```


However, I can do other operations like check if index exists, serahc, delete index successfully.

**Comment by arushijain** [comment]
same for putMapping 

```
  val res = client.execute {
             putMapping("test" / "info").as(
               dateField("search_time"),
               dateField("metadata.checkin_date")
             )
         }.await


java.lang.NoSuchMethodError: com.sksamuel.elastic4s.indexes.PutMappingBuilder$.apply(Lcom/sksamuel/elastic4s/mappings/PutMappingDefinition;)Lorg/elasticsearch/common/xcontent/XContentBuilder;
  at com.sksamuel.elastic4s.http.index.IndexAdminImplicits$PutMappingExecutable$.execute(IndexAdminImplicits.scala:215)
  at com.sksamuel.elastic4s.http.index.IndexAdminImplicits$PutMappingExecutable$.execute(IndexAdminImplicits.scala:203)
  at com.sksamuel.elastic4s.http.HttpClient$class.execute(HttpClient.scala:23)
  at com.sksamuel.elastic4s.http.HttpClient$$anon$1.execute(HttpClient.scala:37)
  ... 47 elided
```

This is most def a Elastic4s Core versioning issue. Which version do I need to be on to have access to this functionality?

**Comment by sksamuel** [comment]
This should be fixed in the latest 5.4, 5.5 and 5.6 branches.
